src/components/pages/weather/elements/WeatherHeadingCell.js

```diff
@@ -2,13 +2,19 @@ import React, { Component } from 'react';
 
 export default class WeatherHeadingCell extends Component {
   render() {
+    const season = (this.props.season ? (this.props.season) : (null));
     const date = new Date(this.props.earthDate);
 
-    return (
+    return ((!this.props.earthDate) ? (
+      <div className="weather-cell">
+        <h4>unknown date</h4>
+        <p>({season})</p>
+      </div>
+    ) : (
       <div className="weather-cell">
         <h4>{date.getDate()}/{date.getMonth()}/{date.getFullYear()}</h4>
-        <p>({this.props.season})</p>
+        <p>({season})</p>
       </div>
-    );
+    ));
```

Comment:
In general we only need parentheses around ternary operator elements when it's ambiguous or confusing which parts we want to evaluate - there's a lot of debate among programmers what that means, but in this case I think we can remove the brackets around the entire statement and around `this.props.earthDate` without any issues.

Parentheses are a tricky one - sometimes they make code less confusing and sometimes they can end up making it cluttered and harder to read!

Reply:
Another general practice is avoiding the **NOT** operator (!) when a statement could be written equally simply without it by reversing the conditions, e.g:
```
const parity = !isEven ? 'odd' : 'even';

const parity = isEven ? 'even' : 'odd';
```
The statements are logically equivalent, but the second one is easier to parse as you don't have to do any logical inversions in your brain when reading it.